### PR TITLE
fixing shuffling + restarting player on error issues

### PR DESCRIPTION
**Pull request by lrlna**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,7 +3,9 @@ var Point = require("point");
 var Emitter = require("emitter");
 
 function getRandomElement (items) {
-  return items[Math.floor(Math.random()*items.length)];
+  var rando = items[Math.floor(Math.random()*items.length)];
+  console.log("random item " + rando);
```
Comment (jb55) [inline index.js]: remove log here, also I have a component that does this https://github.com/jb55/pickrand (its lame but surprisingly common)

File: index.js [hunk 2]
```diff
@@ -257,16 +274,30 @@ AudioPlayer.prototype.getUnplayed = function () {
       arr.splice(index, 1);
     }
   }
+  console.log("array " + arr);
```
Comment (jb55) [inline index.js]: remove log here

**Comment by jb55** [comment]
LGTM after a quick skim, thanks for adding tests!

File: index.js (hunk 2 above)
Comment (lrlna) [inline index.js]: ah shit logs. kk

**Comment by tmathews** [comment]
👍

**Pull request merged**